docs/qml/ml_tools.md

```diff
@@ -209,3 +209,76 @@ for i in range(n_epochs):
     loss.backward()
     optimizer.step()
 ```
+
+
+## Custom `train` loop
+
+If you need custom training functionality that goes beyon what is available in
+`qadence.ml_tools.train_with_grad` and `qadence.ml_tools.train_gradient_free` you can write your own
+training loop based on the building blocks that are available in Qadence.
+
+A simplified version of Qadence's train loop is defined below. Feel free to copy it and modify at
+will.
+
+```python
+from typing import Callable, Union
+
+from torch.nn import Module
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+from torch.utils.tensorboard import SummaryWriter
+
+from qadence.ml_tools.config import TrainConfig
+from qadence.ml_tools.data import DictDataLoader, data_to_device
+from qadence.ml_tools.optimize_step import optimize_step
+from qadence.ml_tools.printing import print_metrics, write_tensorboard
+from qadence.ml_tools.saveload import load_checkpoint, write_checkpoint
+
+
+def train(
+    model: Module,
+    data: DataLoader,
+    optimizer: Optimizer,
+    config: TrainConfig,
+    loss_fn: Callable,
+    device: str = "cpu",
+    optimize_step: Callable = optimize_step,
+    write_tensorboard: Callable = write_tensorboard,
+) -> tuple[Module, Optimizer]:
+
+    # Move model to device before optimizer is loaded
+    model = model.to(device)
```

Comment:
would it be an idea in general to have `device = None` so that if the user happen to have already put everything where they want it we don't make any changes to that? 

Or would that just be more complicated and error prone for 99.99% of users?

Reply:
the idea of our training loop was to cover something like 80% of use cases and everything that goes beyond can be implemented by users themselves so that we don't end up with a `train` loop beast in qadence. this is why I want to add this example of a simplified train loop to the docs. If people want to take care of device things before `train`, the can remove the two related lines in this template:)

Reply:
I do agree, but at the same time I wonder if it could potentially fix a bug in the future.
On the other hand it could introduce bugs too... so hard to say.

LGTM